upf/upf_pfcp.c

```diff
@@ -794,6 +796,9 @@ pfcp_make_pending_far (upf_session_t * sx)
 	if (old->forward.flags & FAR_F_REDIRECT_INFORMATION)
 	  cpy_redirect_information (&new->forward.redirect_information,
 				    &old->forward.redirect_information);
+	if (old->forward.flags & FAR_F_FORWARDING_POLICY)
+	  new->forward.forwarding_policy.identifier =
+	    vec_dup (old->forward.forwarding_policy.identifier);
```

Comment:
This together with the unref in pfcp_update_apply will screw up the reference counters in the forward policy.

Please add a test case that applies a FAR update and verifies that the reference counters are correct.